src/Streamly/Internal/Data/Stream/Channel/Type.hs

```diff
@@ -619,7 +625,7 @@ _getBound = _bound
 -- simpler code.
 --
 setReleaseCb :: AllocateIO -> Config -> Config
-setReleaseCb f cfg = cfg { _release = Just (register f) }
+setReleaseCb f cfg = cfg { _release = Just (registerWith f Priority1) }
```

Comment:
The callback registered in the channel config will always be called first, and then the other callbacks.
The user has to be wary about resource cleanup and decide what should go where.

Reply:
only streamly developers need to worry about the priority. For normal users there is one priority.